crates/core/src/types.rs

```diff
@@ -89,15 +90,23 @@ impl TryFrom<Bundle> for ParsedBundle {
             })
             .collect::<Result<Vec<Recovered<OpTxEnvelope>>, String>>()?;
 
+        let uuid = bundle
+            .replacement_uuid
+            .clone()
+            .unwrap_or_else(|| Uuid::new_v4().to_string());
+
+        let uuid = Uuid::parse_str(uuid.as_str()).map_err(|_| format!("Invalid UUID: {uuid}"))?;
```

Comment:
Oh, this isn't what I thought was going to happen. I thought `replacement_uuid` was going to change from an `Option<String>` to a `Option<Uuid>`. Then `AcceptedBundle::new()` would populate its uuid field with `replacement_uuid.unwrap_or_else(|| Uuid::new_v4())` (without the `to_string`).

If `ParsedBundle` has its own `uuid` field, the same bundle is going to get multiple uuids assigned over its lifetime.